plugins/src/gcc/imp.rs

```diff
@@ -0,0 +1,1333 @@
+
+        let mut list = gst::BufferList::new();
+        let mutlist = list.get_mut().unwrap();
+
+        // Leak the bucket so it can hold at most 30ms of data
+        let maximum_remaining_bits = 30. * self.estimated_bitrate as f64 / 1000.;
+        while (budget > 0 || remaining > maximum_remaining_bits) && !self.buffers.is_empty() {
+            let buf = self.buffers.pop_back().unwrap();
+            let n_bits = buf.size() * 8;
+
+            mutlist.add(buf);
+            budget = budget.saturating_sub(n_bits as i64);
+            remaining -= n_bits as f64;
+        }
+
+        gst::trace!(
+            CAT,
+            obj: bwe,
+            "{} bitrate: {}ps budget: {}/{} sending: {} Remaining: {}/{}",
+            pdur(&elapsed),
+            human_kbits(self.estimated_bitrate),
+            human_kbits(budget as f64),
+            human_kbits(total_budget as f64),
+            human_kbits(list.calculate_size() as f64 * 8.),
+            human_kbits(remaining),
+            human_kbits(total_size)
+        );
+
+        self.last_push = now;
+
+        list
+    }
+
+    fn compute_increased_rate(&mut self, bwe: &super::BandwidthEstimator) -> Option<Bitrate> {
+        let now = time::Instant::now();
+        let target_bitrate = self.target_bitrate_on_delay as f64;
+        let effective_bitrate = self.detector.effective_bitrate();
+        let time_since_last_update_ms = match self.last_increase_on_delay {
+            None => 0.,
+            Some(prev) => {
+                if now - prev < *DELAY_UPDATE_INTERVAL {
+                    return None;
+                }
+
+                (now - prev).as_millis() as f64
+            }
+        };
+
+        if effective_bitrate as f64 - target_bitrate as f64 > 5. * target_bitrate / 100. {
+            gst::info!(
+                CAT,
+                "Effective rate {} >> target bitrate {} - we should avoid that \
+                 as much as possible fine tuning the encoder",
+                human_kbits(effective_bitrate),
+                human_kbits(target_bitrate)
+            );
+        }
+
+        self.last_increase_on_delay = Some(now);
+        if self.ema.estimate_is_close(effective_bitrate) {
+            let bits_per_frame = target_bitrate / 30.;
+            let packets_per_frame = f64::ceil(bits_per_frame / (1200. * 8.));
+            let avg_packet_size_bits = bits_per_frame / packets_per_frame;
+
+            let rtt_ms = self.detector.rtt().num_milliseconds() as f64;
+            let response_time_ms = 100. + rtt_ms;
+            let alpha = 0.5 * f64::min(time_since_last_update_ms / response_time_ms, 1.0);
+            let threshold_on_effective_bitrate = 1.5 * effective_bitrate as f64;
+            let increase = f64::max(
+                1000.0f64,
+                f64::min(
+                    alpha * avg_packet_size_bits,
+                    // FIXME: Until we implement "stuffing" which should ensure that the effective bitrate is
```

Comment:
Where are we on the stuffing front? :)

Reply:
[It is implemented in ](https://github.com/centricular/webrtcsink/pull/66/commits/8836480667368204dd6258a279981363b44054b2)

Reply:
right, my point was that if it is implemented, then shouldn't we do what the comment says and remove it? Sorry for replying only now, but you marked the discussion as resolved ..

Reply:
The comment was removed as it wasn't making sense, which is why the comment was discussed is resolved.

Reply:
What do you mean the comment was removed? It is clearly still part of the merge request, or I'm going blind

Reply:
I am blind :/ I must have expanded the wrong `mod.rs` in the github UI. Sorry about that.

I reworked the comment now so that we keep the behavior in any case, it shouldn't happen anymore when stuffing is on, but still.